rust/core/src/serialization.rs

```diff
@@ -187,57 +203,69 @@ pub trait SerializeEmbeddedGroup {
     ) -> cbor_event::Result<&'a mut Serializer<W>>;
 }
 
+pub trait Deserialize {
+    fn deserialize<R: BufRead + Seek>(
```

Comment:
These changes can be backported to cddl-codegen so that we can generate the appropriate code for the WASM wrappers since traits don't work in WASM.

Do we want `to_bytes()` and `to_cbor_bytes()` for all WASM wrappers? We only had `to_bytes()` before but I felt it was potentially confusing as in the old code this was equivalent to the CBOR binary spec *most* of the time, but not always. This way it should be more obvious.